api/tests/Unit/PoolPolicyTest.php

```diff
@@ -0,0 +1,299 @@
+        $this->assertTrue($this->requestResponderUser->can('viewAnyPublishedAdvertisement', Pool::class));
+        $this->assertTrue($this->adminUser->can('viewAnyPublishedAdvertisement', Pool::class));
+    }
+
+    /**
+     * Assert that the following can view a pool:
+     *
+     * pool operator, platform admin
+     *
+     * @return void
+     */
+    public function testView()
+    {
+        $this->assertTrue($this->poolOperatorUser->can('view', $this->teamPool));
+        $this->assertTrue($this->adminUser->can('view', $this->teamPool));
+
+        $this->assertFalse($this->guestUser->can('view', $this->teamPool));
+        $this->assertFalse($this->applicantUser->can('view', $this->teamPool));
+        $this->assertFalse($this->requestResponderUser->can('view', $this->teamPool));
+        // Pool operator cannot view other teams pools
+        $this->assertFalse($this->poolOperatorUser->can('view', $this->unOwnedPool));
+    }
+
+    /**
+     * Assert that the following can view a draft pool advertisement:
+     *
+     * pool operator, platform admin
+     *
+     * @return void
+     */
+    public function testViewDraftAdvertisement()
+    {
+        $this->teamPool->published_at = null;
+        $this->teamPool->save();
+
+        $this->assertTrue($this->poolOperatorUser->can('viewAdvertisement', $this->teamPool));
+        $this->assertTrue($this->adminUser->can('viewAdvertisement', $this->teamPool));
+
+        $this->assertFalse($this->guestUser->can('viewAdvertisement', $this->teamPool));
+        $this->assertFalse($this->applicantUser->can('viewAdvertisement', $this->teamPool));
+        $this->assertFalse($this->requestResponderUser->can('viewAdvertisement', $this->teamPool));
+
+        // Pool operator cannot view other teams draft pools
+        $this->unOwnedPool->published_at = null;
+        $this->unOwnedPool->save();
+
+        $this->assertFalse($this->poolOperatorUser->can('viewAdvertisement', $this->unOwnedPool));
+    }
+
+    /**
+     * Assert that anyone can view a published pool advertisement:
+     *
+     * @return void
+     */
+    public function testViewPublishedAdvertisement()
+    {
+        $this->teamPool->published_at = config('constants.past_date');
+        $this->teamPool->save();
+
+        $this->assertTrue($this->guestUser->can('viewAdvertisement', $this->teamPool));
+        $this->assertTrue($this->applicantUser->can('viewAdvertisement', $this->teamPool));
+        $this->assertTrue($this->poolOperatorUser->can('viewAdvertisement', $this->teamPool));
+        $this->assertTrue($this->requestResponderUser->can('viewAdvertisement', $this->teamPool));
+        $this->assertTrue($this->adminUser->can('viewAdvertisement', $this->teamPool));
+    }
+
+    /**
+     * Assert that only pool operators can create pools
+     *
+     * @return void
+     */
+    public function testCreate()
+    {
+        $this->assertTrue($this->poolOperatorUser->can('create', Pool::class));
```

Comment:
I can't find any way of passing request data to the policy checks here and am unsure how to get this test working properly.